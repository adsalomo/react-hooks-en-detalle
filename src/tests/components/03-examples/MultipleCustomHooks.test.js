import React from 'react';
import { shallow } from 'enzyme';

import '@testing-library/jest-dom';
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';
import { useFetch } from '../../../hooks/useFetch';
import { useCounter } from '../../../hooks/useCounter';

jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');

describe('Prueba MultipleCustomHooks', () => {

    useCounter.mockReturnValue({
        counter: 10
    });

    test('Debe mostrarse correctament', () => {
        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        });

        const wrapper = shallow(<MultipleCustomHooks />);
        expect(wrapper).toMatchSnapshot(wrapper);
    });

    test('Debe de mostrar la información', () => {
        useFetch.mockReturnValue({
            data: [
                {
                    author: 'Adrián',
                    quote: 'Hola mundo'
                }
            ],
            loading: false,
            error: null
        });

        const wrapper = shallow(<MultipleCustomHooks />);
        //console.log(wrapper.html());
        // buscamos por las clases
        expect(wrapper.find('.alert').exists()).toBe(false);
        expect(wrapper.find('.mb-0').text().trim()).toBe('Hola mundo');
        expect(wrapper.find('footer').text().trim()).toBe('Adrián');
    });


});
