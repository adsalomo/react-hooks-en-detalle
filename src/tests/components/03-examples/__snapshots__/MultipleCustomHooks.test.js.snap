// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba MultipleCustomHooks Debe mostrarse correctament 1`] = `Object {}`;
